### PR TITLE
Adds React 18 support

### PR DESCRIPTION
**Pull request by chrissrogers**
- ⚠️ **This includes breaking changes** ⚠️ Subsequent releases must bump the major version
  - peerDependency `react` and `react-dom` increased from `16.x` to `18.x`
- Bumps React major version from 16 to 18
- Removes `lodash/cloneDeep` dependency in favor of native `structuredClone`
- Updates demo React implementation
- Fully re-implements documentation for storybook bump from `5.x` to `7.x` and major changes to their story and docs composition
- Re-implements some tests to remove `enzyme` usage in favor of `@testing-library/react`

### CODE REVIEW

**Comment by chrissrogers** [comment]
Fixes #186 and #175

File: lib/provider.jsx [hunk 1]
```diff
@@ -85,7 +85,7 @@ export default class RecurlyProvider extends React.Component {
   render() {
     return (
       <Provider value={{ recurly: this._recurly }}>{this.props.children}</Provider>
-    )
+    );
```
Comment (jorge) [inline lib/provider.jsx:88]: L G T M

File: lib/use-checkout-pricing.js
Comment (jorge) [inline lib/use-checkout-pricing.js:1]: why wasn't this one renamed to `jsx` too?

**Review by douglasmiller — APPROVED** [review]

File: lib/use-checkout-pricing.js
Comment (chrissrogers) [reply lib/use-checkout-pricing.js:1]: No reason really. I've updated it to go along, but it's more about informing editor syntax than it is any build behavior.

Comment (chrissrogers) [reply lib/use-checkout-pricing.js:1]: The filename doesn't inform any different processing in the build and is just for editor syntax hints. This file in particular doesn't contain any JSX, just plain JS.

**Pull request merged**